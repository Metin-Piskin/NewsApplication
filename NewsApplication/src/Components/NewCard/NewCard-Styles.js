import { StyleSheet } from "react-native";


export default StyleSheet.create({
    allcontainer: {
        //borderWidth: 1,
        marginTop: 10,
        maxWidth: 375,
        marginHorizontal: 5,
        borderRightWidth: 10,
        borderBottomWidth: 10,
        marginLeft: 10,
        backgroundColor: '#e0e0e0'
    },
    image: {
        //width: Dimensions.get('screen').width,
        maxWidth: 352,
        height: 270,
        resizeMode: "contain",
        marginHorizontal: 5,
        marginVertical: 5,
        justifyContent: 'flex-end'
    },
    title: {
        paddingHorizontal: 10,
        color: '#fff',
        fontSize: 20,
        backgroundColor: 'rgba(52, 52, 52, 0.8)',
        fontFamily: 'ChakraPetch-MediumItalic'
    },
    content: {
        paddingHorizontal: 5,
        fontFamily: 'ChakraPetch-Bold',
        color: '#000'
    },
    author: {
        color: 'red',
        alignSelf: 'flex-end',
        paddingHorizontal: 5,
        fontFamily: 'ChakraPetch-Bold',
    },
    innercontainer: {
        flexDirection: 'row',
        justifyContent: "space-between",
        paddingHorizontal: 15
    },
    date: {
        fontFamily: 'CourierPrime-Bold',
        color: '#000',
        fontSize:15
    },
    time: {
        fontFamily: 'CourierPrime-Bold',
        color: '#000',
        fontSize:15
    },
})