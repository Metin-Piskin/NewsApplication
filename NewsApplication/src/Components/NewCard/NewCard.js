import React from 'react';
import { View, Text, TouchableOpacity, ImageBackground } from 'react-native';

import styles from './NewCard-Styles';

const NewCard = ({ news, onPress }) => {
    return (
        <>
            {
                news ? (
                    <TouchableOpacity
                        style={styles.allcontainer}
                        onPress={onPress}
                        testID='new-card-press'
                    >
                        <ImageBackground
                            testID='new-card-imagebackground'
                            style={styles.image}
                            source={{ uri: news.imageUrl }}
                        >
                            <Text
                                style={styles.title}
                                testID='new-card-title'
                            >
                                {news.title}
                            </Text>
                        </ImageBackground>
                        <Text
                            style={styles.content}
                            testID='new-card-content'
                        >
                            {news.content}
                        </Text>
                        <Text
                            style={styles.author}
                            testID='new-card-author'
                        >
                            {news.author}
                        </Text>
                        <View
                            style={styles.innercontainer}
                            testID='new-card-innercontainer'
                        >
                            <Text
                                style={styles.date}
                                testID='new-card-date'
                            >
                                {news.date}
                            </Text>
                            <Text
                                style={styles.time}
                                testID='new-card-time'
                            >
                                {news.time}
                            </Text>
                        </View>
                    </TouchableOpacity>

                ) : (
                    <View>
                        <Text>Null</Text>
                    </View>
                )
            }
        </>
    )
}

export default NewCard;