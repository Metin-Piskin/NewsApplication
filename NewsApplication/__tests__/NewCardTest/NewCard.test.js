import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import NewCard from '../../src/Components/NewCard';
import NewCardStyles from '../../src/Components/NewCard/NewCard-Styles';

test('renders correctly', () => {
    const comp = render(<NewCard />);
    expect(comp).toMatchSnapshot();
});

test('renders correctly', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    expect(comp).toMatchSnapshot();
});

test('should render press correctly', () => {
    const test = 'test';
    const mockFunction = jest.fn();
    const comp = render(<NewCard news={test} onPress={mockFunction} />);

    const NewCardPress = comp.getByTestId('new-card-press');
    fireEvent(NewCardPress, 'press');

    expect(mockFunction).toBeCalled();
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardAllContainerStyle = comp.getByTestId('new-card-press').props.style;

    expect(NewCardAllContainerStyle).toMatchObject(NewCardStyles.allcontainer);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardTitleStyle = comp.getByTestId('new-card-title').props.style;

    expect(NewCardTitleStyle).toMatchObject(NewCardStyles.title);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardContentStyle = comp.getByTestId('new-card-content').props.style;

    expect(NewCardContentStyle).toMatchObject(NewCardStyles.content);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardAuthorStyle = comp.getByTestId('new-card-author').props.style;

    expect(NewCardAuthorStyle).toMatchObject(NewCardStyles.author);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardInnerContainerStyle = comp.getByTestId('new-card-innercontainer').props.style;

    expect(NewCardInnerContainerStyle).toMatchObject(NewCardStyles.innercontainer);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardDateStyle = comp.getByTestId('new-card-date').props.style;

    expect(NewCardDateStyle).toMatchObject(NewCardStyles.date);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<NewCard news={test} />);
    const NewCardTimeStyle = comp.getByTestId('new-card-time').props.style;

    expect(NewCardTimeStyle).toMatchObject(NewCardStyles.time);
});