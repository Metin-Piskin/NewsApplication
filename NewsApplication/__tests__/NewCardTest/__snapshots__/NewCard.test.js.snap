// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    Null
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#e0e0e0",
      "borderBottomWidth": 10,
      "borderRightWidth": 10,
      "marginHorizontal": 5,
      "marginLeft": 10,
      "marginTop": 10,
      "maxWidth": 375,
      "opacity": 1,
    }
  }
  testID="new-card-press"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "height": 270,
        "justifyContent": "flex-end",
        "marginHorizontal": 5,
        "marginVertical": 5,
        "maxWidth": 352,
        "resizeMode": "contain",
      }
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 270,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="new-card-imagebackground"
    />
    <Text
      style={
        {
          "backgroundColor": "rgba(52, 52, 52, 0.8)",
          "color": "#fff",
          "fontFamily": "ChakraPetch-MediumItalic",
          "fontSize": 20,
          "paddingHorizontal": 10,
        }
      }
      testID="new-card-title"
    />
  </View>
  <Text
    style={
      {
        "color": "#000",
        "fontFamily": "ChakraPetch-Bold",
        "paddingHorizontal": 5,
      }
    }
    testID="new-card-content"
  />
  <Text
    style={
      {
        "alignSelf": "flex-end",
        "color": "red",
        "fontFamily": "ChakraPetch-Bold",
        "paddingHorizontal": 5,
      }
    }
    testID="new-card-author"
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
      }
    }
    testID="new-card-innercontainer"
  >
    <Text
      style={
        {
          "color": "#000",
          "fontFamily": "CourierPrime-Bold",
          "fontSize": 15,
        }
      }
      testID="new-card-date"
    />
    <Text
      style={
        {
          "color": "#000",
          "fontFamily": "CourierPrime-Bold",
          "fontSize": 15,
        }
      }
      testID="new-card-time"
    />
  </View>
</View>
`;
